agent/consul/state/config_entry_events_test.go

```diff
@@ -547,3 +547,348 @@ func TestServiceDefaultsSnapshot(t *testing.T) {
 		})
 	}
 }
+
+func TestAPIGatewaySnapshot(t *testing.T) {
```

Comment:
🧠  Seems like most of the test code that isn't directly defining `testCases` is duplicated. I'd probably personally just do one big table test with all of the snapshot types, but I see that this code is matching patterns established in this file already.

Reply:
I see what you're saying. I think I'll keep it as-is because it follows the existing pattern. But I would see the value in refactoring this file down the road.